packages/salesforcedx-utils-vscode/test/jest/telemetry/telemetry.test.ts

```diff
@@ -91,4 +91,47 @@ describe('Telemetry', () => {
       expect(result).toBe(telemetryService.extensionName);
     });
   });
+
+  describe('Telemetry Service - getUserId', () => {
+    let telemetryService: TelemetryService;
+    let executeCliTelemetrySpy: jest.SpyInstance;
+    let getRandomUserIdSpy: jest.SpyInstance;
+
+    const fakeCliTelemetryData = {
+      result: {
+        enabled: true,
+        cliId: 'rdccvghuhbnjiokmnmklop'
+      }
+    };
+    const fakeCliIdUndefined = {
+      result: {
+        enabled: true
+      }
+    };
+    const randomId = 'sedtrfyv234dd';
+    beforeEach(() => {
+      telemetryService = new TelemetryService();
+      executeCliTelemetrySpy = jest.spyOn((telemetryService as any), 'executeCliTelemetry');
+      getRandomUserIdSpy = jest.spyOn((telemetryService as any), 'getRandomUserId');
```

Comment:
Should the globalState call be mocked here?

Reply:
mocked get & update function to achieve the same.